src/app/components/CodeStep.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { Button, Form, Input } from 'antd';
+import PropTypes from 'prop-types';
+
+const CodeStep = ({ onSubmitForm }) => {
+  const [form] = Form.useForm();
+  return (
+    <Form form={form} onFinish={onSubmitForm}>
+      <Form.Item
+        name="code"
+        rules={[
+          {
+            required: true,
+            message: 'Please input code!',
+          },
+        ]}
+      >
+        <Input maxLength={25} placeholder="Code" size="large" />
+      </Form.Item>
+      <Form.Item shouldUpdate>
+        {() => (
```

Comment:
Why is it wrapped into function? I guess if you remove wrapper, you will not need **shouldUpdate**